src/main/java/io/confluent/connect/hdfs/wal/WALFile.java

```diff
@@ -208,8 +210,11 @@ public static Option blockSize(long value) {
         long blockSize = blockSizeOption == null ? fs.getDefaultBlockSize(p) :
                          blockSizeOption.getValue();
 
-        if (appendIfExistsOption != null && appendIfExistsOption.getValue()
-            && fs.exists(p)) {
+        if (appendIfExistsOption != null
+            && appendIfExistsOption.getValue()
+            && fs.exists(p)
+            && hasIntactVersionHeader(p, fs)
```

Comment:
I think if `hasIntactVersionHeader` fails, we probably want to log some error level message. As discussed elsewhere, the fact that we hit that condition should be pretty big red flag -- either Connect is randomly dying mid-write or somethings up with your HDFS. At a minimum, a warning-level message.